bottlecap/src/traces/stats_aggregator.rs

```diff
@@ -0,0 +1,82 @@
+use prost::Message;
+use std::collections::VecDeque;
+
+/// Maximum number of entries in a stat payload.
+///
+/// <https://github.com/DataDog/datadog-agent/blob/996dd54337908a6511948fabd2a41420ba919a8b/pkg/trace/writer/stats.go#L35-L41>
+// const MAX_BATCH_ENTRIES_SIZE: usize = 4000;
+
+/// Aproximate size an entry in a stat payload occupies
+///
+/// <https://github.com/DataDog/datadog-agent/blob/996dd54337908a6511948fabd2a41420ba919a8b/pkg/trace/writer/stats.go#L33-L35>
+// const MAX_ENTRY_SIZE_BYTES: usize = 375;
+
+/// Maximum content size per payload in compressed bytes,
+///
+/// <https://github.com/DataDog/datadog-agent/blob/996dd54337908a6511948fabd2a41420ba919a8b/pkg/trace/writer/stats.go#L35-L41>
+const MAX_CONTENT_SIZE_BYTES: usize = 3 * 1024 * 1024; // ~3MB
+
+#[allow(clippy::module_name_repetitions)]
+pub struct StatsAggregator<T>
```

Comment:
Had to make it a <T> because using the `ClientStatsPayload` type directly wouldn't allow me to use the `encoded_len` method from `pb::Message`

Reply:
Seems commit `588585e` fixes this